### ISSUE TITLE
'StartWorkTimer' must be called on the foreground thread.

### ISSUE THREAD

**Issue by eevahr** [aspnet/Razor.VSCode#372]
#### Is this a Bug or Feature request?:

Bug

#### Steps to reproduce

Have razor file open, possibly also be in debugging mode.

#### Description of the problem:

Razor language server stops unexpectedly and without any real indication of what triggers the stop.

Expected behavior:
Razor language server to not stop unexpectedly.

Actual behavior:
Razor language server stops unexpectedly.

razor file/log
https://gist.github.com/eevahr/c25c22612563740b9570a9f8871567fb

**Comment by NTaylorMullen** [comment]
Thanks for the report @eevahr! I wonder what could be going wrong here hmmm. I'll have to take a deeper look.

**Comment by rpaulucci3** [comment]
I would like to second the report. Happens for me on VS Code 1.34.0, C# extension 1.20.0 ,.NET Core SDK 2.2.300. Unfortunately, I can't share stack traces because my company does not allow file uploads.

All I remember is that this exception happens and causes the Razor Language Server to be restarted a lot, crashing immediately with an RpcException every time until VS Code stops restarting it.

**Comment by tachyon1337** [comment]
Same issue. Within the past month or so, VS Code on Mac has repeatedly begun crashing while editing Razor .cshml pages. The stack trace is typically `StartWorkTimer must be called on the foreground thread`

**Comment by rpaulucci3** [comment]
Issue remains in C# Extension 1.21.0.
Happened last time when creating a Razor comment (`@* *@`).

**Comment by NTaylorMullen** [comment]
> Issue remains in C# Extension 1.21.0.
> Happened last time when creating a Razor comment (`@* *@`).

Yup, we haven't had the bandwidth to push out an official fix for this quite yet. Sorry! 😢

**Comment by rpaulucci3** [comment]
Sorry if my previous comment came across as a complaint.

I just meant to inform you that commits in 1.21.0 have not inadvertently fixed the issue, given that you had said you were still looking into the cause of this issue, and that other people in the thread mentioned this might be a regression _("Within the past month or so...")_

That being said, would you be able to share with us if a cause has been identified? (if/when you have time to do so, of course)

**Comment by NTaylorMullen** [comment]
Definitely!

**Comment by nkev** [comment]
@NTaylorMullen If it helps, this exact `JsonRpc.Server` crash occurs reproducably, every time on my Macbook (VS Code and with dotnet Core 2.1) right after I press the "dot" in `Orchard.` as shown in this example:
https://www.youtube.com/watch?v=4o9zG17cfa0&feature=youtu.be&t=1391

**Comment by NTaylorMullen** [comment]
@nkev thank you for the exact info. That'll definitely help in reproducing on my end 😄
